### PR TITLE
chore(gpu): improve performance of comparison operations

### PR DESCRIPTION
**Pull request by pdroalves**
closes: https://github.com/zama-ai/tfhe-rs-internal/issues/436

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by pdroalves** [comment]
@slab-ci integer_gpu_full_bench

**Comment by pdroalves** [comment]
@slab-ci integer_gpu_full_bench

**Comment by pdroalves** [comment]
@slab-ci integer_gpu_full_bench

**Comment by pdroalves** [comment]
@slab-ci integer_gpu_full_bench

**Comment by pdroalves** [comment]
@slab-ci integer_gpu_full_bench

**Comment by pdroalves** [comment]
@slab-ci integer_gpu_full_bench

**Comment by pdroalves** [comment]
@slab-ci integer_gpu_full_bench

**Comment by pdroalves** [comment]
@slab-ci integer_gpu_full_bench

File: backends/tfhe-cuda-backend/cuda/include/integer.h [hunk 1]
```diff
@@ -849,8 +849,7 @@ template <typename Torus> struct int_are_all_block_true_buffer {
   COMPARISON_TYPE op;
   int_radix_params params;
 
-  int_radix_lut<Torus> *is_max_value_lut;
-  int_radix_lut<Torus> *is_equal_to_num_blocks_lut;
+  std::unordered_map<int, int_radix_lut<Torus> *> is_equal_to_num_blocks_map;
```
Comment (agnesLeroy) [inline backends/tfhe-cuda-backend/cuda/include/integer.h:852]: Name: `is_equal_to_lut_map`
Could you add a comment above saying that this map contains two int_radix_luts, one to compare equality with max_value (the maximum message value) and one to compare equality with num_blocks.

File: backends/tfhe-cuda-backend/cuda/src/integer/integer.cuh [hunk 2]
```diff
@@ -11,6 +11,7 @@
 #include "polynomial/functions.cuh"
 #include "utils/kernel_dimensions.cuh"
 #include <functional>
+#include <omp.h>
```
Comment (agnesLeroy) [inline backends/tfhe-cuda-backend/cuda/src/integer/integer.cuh:14]: This is not impacting performance so let's remove omp from this

**Comment by pdroalves** [comment]
Thanks for the review @agnesLeroy ! I've just pushed the changes.

**Review by agnesLeroy — APPROVED** [review]

**Comment by pdroalves** [comment]
Thanks for the review @agnesLeroy ! I've just pushed the changes.

**Pull request merged**